crates/bevy_core_pipeline/src/prepass/node.rs

```diff
@@ -0,0 +1,145 @@
+use bevy_ecs::prelude::*;
+use bevy_ecs::query::QueryState;
+use bevy_render::{
+    camera::ExtractedCamera,
+    prelude::Color,
+    render_graph::{Node, NodeRunError, RenderGraphContext, SlotInfo, SlotType},
+    render_phase::{DrawFunctions, RenderPhase, TrackedRenderPass},
+    render_resource::{
+        LoadOp, Operations, RenderPassColorAttachment, RenderPassDepthStencilAttachment,
+        RenderPassDescriptor,
+    },
+    renderer::RenderContext,
+    view::{ExtractedView, ViewDepthTexture},
+};
+#[cfg(feature = "trace")]
+use bevy_utils::tracing::info_span;
+
+use super::{AlphaMask3dPrepass, Opaque3dPrepass, ViewPrepassTextures};
+
+pub struct PrepassNode {
+    main_view_query: QueryState<
+        (
+            &'static ExtractedCamera,
+            &'static RenderPhase<Opaque3dPrepass>,
+            &'static RenderPhase<AlphaMask3dPrepass>,
+            &'static ViewDepthTexture,
+            &'static ViewPrepassTextures,
+        ),
+        With<ExtractedView>,
+    >,
+}
+
+impl PrepassNode {
+    pub const IN_VIEW: &'static str = "view";
```

Comment:
It's used when inserting it into the render graph, this pattern is used for all nodes in the main_core_pipeline. I agree this should be documented, but I think it might need a more centralized place to document the pattern.